src/components/TodoForm/TodoForm.tsx

```diff
@@ -0,0 +1,90 @@
+import React, { useState } from 'react';
+import usersFromServer from '../../api/users';
+import { Todo } from '../../App';
+
+type Props = {
+  todos: Todo[];
+  updateTodoList: React.Dispatch<React.SetStateAction<Todo[]>>;
+};
+
+export const TodoForm: React.FC<Props> = ({ todos, updateTodoList }) => {
+  const [currentUserId, setCurrentUserId] = useState(0);
+  const [title, setTitle] = useState('');
+  const [isButtonClicked, setIsButtonClicked] = useState(false);
+
+  const addedTodoId = Math.max(...todos.map(todo => todo.id)) + 1;
+  const preparedObject = {
+    id: addedTodoId,
+    title: title,
+    completed: false,
+    userId: currentUserId,
+  };
+
+  const isUserSelected = () => (currentUserId ? true : false);
+  const isTitleValid = () => (title ? true : false);
+
+  return (
+    <form action="/api/todos" method="POST">
+      <div className="field">
+        <label htmlFor="title">Title: </label>
+        <input
+          id="title"
+          type="text"
+          data-cy="titleInput"
+          placeholder="Enter a title"
+          value={title}
+          onChange={event => setTitle(event.target.value)}
+        />
+
+        {!isTitleValid() && isButtonClicked && (
+          <span className="error">Please enter a title</span>
+        )}
+      </div>
+
+      <div className="field">
+        <label htmlFor="user">User: </label>
+        <select
+          id="user"
+          data-cy="userSelect"
+          value={currentUserId}
+          onChange={event => setCurrentUserId(+event.target.value)}
+        >
+          <option value="0" disabled>
+            Choose a user
+          </option>
+          {usersFromServer.map(user => {
```

Comment:
Ensure that `usersFromServer` is an array of user objects. If `usersFromServer` is undefined or not an array, it might cause issues when mapping over it.